### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by dhdaines**
Various documentation improvements

### CODE REVIEW

**Comment by dhdaines** [comment]
Hi! Here's a preliminary cleanup of the documentation to:

- Add an introduction with a lot of useful links to lower-level pages
- Fill in missing descriptions
- Make the descriptions of algorithms more consistent
- List a number of differences from OpenFST that are useful to know

File: rustfst/src/algorithms/optimize.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ use crate::semirings::{SemiringProperties, WeaklyDivisibleSemiring, WeightQuanti
 use crate::Semiring;
 use anyhow::Result;
 
+/// General optimization (determinization and minimiaztion) of a WFST
```
Comment (Garvys) [inline rustfst/src/algorithms/optimize.rs:10]: typo

File: rustfst/src/algorithms/inversion.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@ use crate::fst_properties::FstProperties;
 use crate::fst_traits::MutableFst;
 use crate::semirings::Semiring;
 
-/// This operation inverts the transduction corresponding to an FST
+/// Inverts the transduction corresponding to an FST
```
Comment (Garvys) [inline rustfst/src/algorithms/inversion.rs:6]: `Invert`

**Review by Garvys — CHANGES_REQUESTED** [review]
Very nice !!! Thanks for taking the time of doing this ! 
Small typos and then I'm good for a merge!
(Sorry for the late answer, missed the notification 😬 )

**Comment by dhdaines** [comment]
Done! I will probably write some more, will make another PR for those later.

**Comment by Garvys** [comment]
Great !! Thanks so much 💪

**Review by Garvys — APPROVED** [review]
🚀

**Pull request merged**